storage/innobase/srv/srv0srv.cc

```diff
@@ -3542,6 +3544,15 @@ srv_enable_redo_encryption()
 bool
 srv_enable_redo_encryption_mk()
 {
+	if (existing_redo_encryption_mode != REDO_LOG_ENCRYPT_MK
+	    && existing_redo_encryption_mode != REDO_LOG_ENCRYPT_OFF)
+	{
+		ib::warn() << "Redo log encryption mode can't be switched without "
```

Comment:
I think SET command should throw an error. Not the ib::error(), Send error to client. Can you please check this.

Also startup should fail with clear message. "Current redo logs are encrypted with MK. Please start with innodb_log_encrypt=MK to continue startup with MK Or delete redo logs blah blah.."

Reply:
Everything else redo encryption related only results in a warning and continues without doing anything - including the server start. (and this is upstream behavior, we shouldn't change it)

I'll update the sending to the client part.

Reply:
well, upstream doesn't have two types, isn't it? So can you please explain what you mean by upstream behaviour here?

Reply:
I mean every other possible problem with redo encryption. Like the test case I've added in my previous PR - if there is no keyring, we display a warning and continue without encryption.
That's how basically any error handling works in redo encryption, and unless the server actually ends up doing recovery, it continues to run.